src/core/demangle.d

```diff
@@ -2565,6 +2564,17 @@ version(unittest)
     }
 }
 
+unittest
+{
+    // https://issues.dlang.org/show_bug.cgi?id=18300
+    string s = demangle.mangleof;
+    foreach(i; 1..77)
+    {
+        char[] buf = new char[i];
+        auto ds = demangle(s, buf);
+        assert(ds == "pure nothrow @safe char[] core.demangle.demangle(const(char)[], char[])");
+    }
+}
```

Comment:
Nice! Here's another way to test for big symbols:

```
import core.demangle, std.stdio;
void main()
{
    string s = "_D1";
    foreach (_; 0..10_000)
        s ~= "a1";
    s ~= "FiZi";
    char[] dst = new char[10_000];
    demangle(s, dst).writeln;
}
```

https://run.dlang.io/is/fJ9T61